feat: add siliconflow chat provider

qychen2001 opened the pull request:
Users can now configure SiliconFlow like any other OpenAI-compatible backend, pick models via the UI, store keys safely, and stream/function-call through the existing chat pipeline. Embedding/vector support remains a future follow-up.

## Checklist:

- [x] I have read the [Contribution guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

qychen2001 committed:
feat: add siliconflow chat provider

Cohee1207 reviewed (changes requested):
1. Seed parameter will not work. It requires a separate handling in the code: `seedSupportedSources` array in `sendOpenAIRequest`
2. This PR omits prompt image inlining support. Does the API support it? If yes, add to `isImageInliningSupported` and unlock the UI control.
3. There is a missing block of code that handles API context size change in `onModelChange`
4. Reasoning controls are hidden in the HTML, you won't be able to change reasoning effort or toggle reasoning display.
5. Reasoning parsing is not implemented in the code for streaming/non-streaming.

qychen2001 committed:
Merge branch 'staging' into feat-add-siliconflow

qychen2001 committed:
fix: remove siliconflow seed handling

qychen2001 committed:
fix: enable SiliconFlow image inlining from shared vision list

qychen2001 committed:
fix: handle SiliconFlow model context limits and sizing

qychen2001 committed:
fix: drop SiliconFlow reasoning effort handling since controls are hidden

qychen2001 committed:
fix: parse SiliconFlow reasoning in streaming and non-streaming

qychen2001 committed:
fix: add missing comma in supported models list

qychen2001 commented:
1. I deleted the seed parameter because it is not supported by the API
2. I implemented this feature, as shown in the image below
<img width="1894" height="1190" alt="image" src="https://github.com/user-attachments/assets/8a7d594a-a006-4515-8bda-94da259be79f" />
3. Fixed, now when the model changes, the context size will also change, as shown in the image
<img width="934" height="416" alt="image" src="https://github.com/user-attachments/assets/2b60edf7-2644-492b-bbf9-fdb7e5a9981e" />
4. The API of siliconflow does not support reasoning effort, it has been deleted
5. Reasoning can now be displayed on the frontend, as shown in the image
<img width="1906" height="640" alt="image" src="https://github.com/user-attachments/assets/d341bdd5-5aae-4ac0-b705-b9558f0a0ecd" />

Thank you for your detailed review of my code. I hope my changes have addressed most of your concerns, and I look forward to merging.

Cohee1207 committed:
Merge branch 'staging' into feat-add-siliconflow

Cohee1207 committed:
Remove seed from HTML template

Cohee1207 committed:
Combine streamed reasoning parsing blocks

Cohee1207 committed:
Update logo image

Cohee1207 reviewed (approved):
I checked it, this one is good to go. Let me know when you decide to add other goodies like image generation, vector storage, image captioning, TTS, etc.

Cohee1207 commented:
Some further considerations I see after checking the API docs for this one @qychen2001 
1. Min P param is not mapped. It's supported by Chat Completions in SillyTavern, you just need to manually pass it to the backend in request payload. 
2. Same as above, but with Top K.
3. N parameter is unmapped. In ST it corresponds to "multiple swipes per generation" feature (still uses `n` param).
4. `stop` - didn't check this one, but you can pass custom stop strings from ST's advanced formatting to the API. Check how it's done for other providers.
5. `reasoning_effort` (ST) can be mapped to `thinking_budget` (API) as it's currently done for Gemini and Claude sources. Could use different ratios of relation to max response tokens (i.e. 128 for minimum, x0.2 for low, etc.)
6. `enable_thinking` (API) can be mapped to `show_thoughts` (ST). I know it says "for visibility only" but it's currently not the case for ZAI, and that text would need tweaking as it _sometimes_ affects the actual request.

Neither of those are critical to be fixed now and can be addressed in a follow up PR. Let me know if you have any additional questions or require assistance with any of the points (I know it's not a perfect code base to work with).

Cohee1207 committed:
Unhide image inlining controls
